### ISSUE TITLE
After infant update mod throws BE's every few seconds

### ISSUE THREAD

**Issue by kelliq** [TitanNano/TS4ControlAnySim#67]
I removed the mod and didnt save any of the BE's but I wanted to bring it to your attention anyways. The mod did seem to function, what little I could test of it anyway, I couldnt really play because it was throwing out errors every 5 to 10 seconds so I couldnt get much testing done between exiting out of the constant error messages and BE scans. I got annoyed and removed the mod.

**Comment by TitanNano** [comment]
Unfortunately, I need at least one of the BEs to be able to investigate this. Please add the mod into your game again and attach all the BEs that you can get to this issue.

**Comment by kelliq** [comment]
Im not getting the same error reports that I was getting this time around. The error reports are no longer pointing to control any sim as 100% the cause, they are now saying cpp_message (whatever that is). I will attach one anyway. 
![Screenshot (183)](https://user-images.githubusercontent.com/131380656/233695311-b375a31e-57d7-4586-a9f5-c333ece128b1.png)
It wouldnt allow me to attach it so I had to do a screenshot

**Comment by TitanNano** [comment]
Please upload the report file here. Your screenshot appears to indicate that the error you are seeing is not related to this mod.

**Comment by TitanNano** [comment]
@G3RGEOUS can you please provide an actual exception that you are seeing? This file is unfortunately not of much use.

**Comment by Pirakeet** [comment]
> @G3RGEOUS can you please provide an actual exception that you are seeing? This file is unfortunately not of much use.

Not identical issue, but didn't want to open a new one due to likely false positive/user error/education gap on my end.  

Attached formatted BE and rawLastException. Can send the full BE html if I can send privately (so sorry). 

First time with the reference of holidays, but I get a lot of ''AttributeError: 'NoneType' object has no attribute'' and exceptions related to ''scheduler.py" that often reference ControlAnySim, TimelineUnclogger & SimulationLagFix. Happy Haunts has recently started showing up though.

**Any extra info I can think of:**

**Control Any Sim:**    Not currently in use at time of exception.
**Holiday:**   Custom Holiday including custom traditions
**Sim in use:**    Just cheated into family via (debug menu?)
Both locations not utilising Happy Haunts 

**Actions at time of exception:**  Travel from home lot to owned retail business (Newcrest). BE Started generating and crash to worlds menu. Unable to re-enter lot and active sim not present in family thumbnails while attempting.
**Actions taken:** New save file, reboot game. Active sim back and onsite at retail lot. 

If there's anything with a clear cause that is from a different mod, let me know and I can let them know - but otherwise just an FYI in my case. 

Hope this helps! Thank you for a fantastic mod!


![Control any sim 1](https://github.com/TitanNano/TS4ControlAnySim/assets/138599968/25b7352e-e571-47a1-ade3-7dcacd4bf2a2)
[ControlAnySim Formatted BE.txt](https://github.com/TitanNano/TS4ControlAnySim/files/11950287/ControlAnySim.Formatted.BE.txt)
[RawLastException-20230705-002702.txt](https://github.com/TitanNano/TS4ControlAnySim/files/11950291/RawLastException-20230705-002702.txt)

**Comment by TitanNano** [comment]
@Pirakeet looking at the exception, it is very unlikely that this is caused by ControlAnySim. The reason the mod is mentioned, is because the mod hooks into some very basic gameplay logic. Can you reproduce this exception with only the ControlAnySims mod? It's more likely that this is caused by something holidays related.

**Comment by Pirakeet** [comment]
@TitanNano  Got another one for you, unrelated to the holiday one - but actively family is the same. Moved into a brand new save file, but I have no idea how much broken I can port over lol. No holiday active.

"Exception occurred while attempting to populate a repeated field: (AttributeError: 'NoneType' object has no attribute 'missing_pet_tracker')"

Attached info and attempted to breakdown my actions - let me know if more info required. 

Reconfirming: for your reference, no support required on my end. 

**Event:** 
New save - newcrest - empty lot.
Had visited a household of 8 (all also imported) and used MCCC to update all relationships with active household. Controlled all 8 additional sims via ControlAnySim. 

Moved 1 sim to new household (unknown if I ''stopped controlling sim)

Stopped controlling 2 more sims, moved 1  sim into a new household via MCCC.
Moved another sim into the above household via MCCC.
Exception popup occured.
Still have control of the other sims. 



[RawLastException-20230712-053237.txt](https://github.com/TitanNano/TS4ControlAnySim/files/12018454/RawLastException-20230712-053237.txt)
![Formatted LastException Screenshot](https://github.com/TitanNano/TS4ControlAnySim/assets/138599968/20b38ebf-99e7-482c-bd7a-c8c1c2578a7b)
![exception summary screenshot](https://github.com/TitanNano/TS4ControlAnySim/assets/138599968/6ded1d74-ad2b-477d-b024-2ce586060caa)
[Formatted LastException.txt](https://github.com/TitanNano/TS4ControlAnySim/files/12018501/Formatted.LastException.txt)

**Comment by TitanNano** [comment]
This exception is actually related to this mod. It occurred when taking control of a sim. 

Now it would be good to know what made this sim special as the game is looking for certain features on the sim that are not present. Do you have any clue which sim you were taking control of at the time?

**Comment by Pirakeet** [comment]
Apologies for the delay, I will update my email.

I don't believe I can identify the exact sim but I can narrow it down to 4-5 sims - in 2 households of 8. I'm not sure if that's too large an undertaking on your end. I will also try to replicate the exception to narrow it down.

Regardless, let me know what you need (If it's sim data more complex than MCCC detailed log output and it's not easy to google, I might need more info on extracting). Tray/Saves/Gallery is also fine. 

(P.S I just followed you on twitter (same handle) for backup contact if I ghost you and you're wanting to follow up. Only if it's bugging you, of course)

**Comment by Pirakeet** [comment]
Okay, so I was able to recreate the BE - but I have to assume I was accidentally talking shit because I have now learned the MC Exception is more detailed and specifically referenced the Active and Selected Sim

I had just performed the aforementioned household moves again - but the error occured upon aging up a a newborn and selecting the infant sim

Attempted again with debug-spawned newborn - all ok
Had resident give birth to triplets - all ok

So far it's only the first infant:

**Active Sim:** Bus Driver
**CAS Selected Sim:** Tiffany Driver

**Attached Zip:**
BE-Exceptionreport
lastException.txt
mc_lastexception
MCCC_SimDetails_Bus Driver
MCCC_SimDetails_Tiffany Driver

Let me know if you need anything else!

[CASBE.zip](https://github.com/TitanNano/TS4ControlAnySim/files/12061899/CASBE.zip)

**Comment by Pirakeet** [comment]
I will stop spamming you after this, but I thought it *might* be useful to provide this one, because it's the same exception - presumably the same cause - but different outcomes.

Event 1: Demolished home - newborn went to inventory - placed newborn. Newborn taken away due to neglect. Exception triggers.
Exception cause - baby.py 
Cause: Presumably the zips buried in mod folders. Removed zips and issue didn't recur

Event 2: Reloaded save with new mod test folder. Exception triggers on load
Exception 2 - ControlAnySim
Cause: Probably the buried zips again lol. But CAS identified as the cause.

Both BEs added, lastexception.txt, mc_lastexception.html are both from the second one that identifies CAS as the cause


[baby.zip](https://github.com/TitanNano/TS4ControlAnySim/files/12079774/baby.zip)

**Comment by TitanNano** [comment]
> I will stop spamming you after this, but I thought it _might_ be useful to provide this one, because it's the same exception - presumably the same cause - but different outcomes.
> 
> Event 1: Demolished home - newborn went to inventory - placed newborn. Newborn taken away due to neglect. Exception triggers. Exception cause - baby.py Cause: Presumably the zips buried in mod folders. Removed zips and issue didn't recur
> 
> Event 2: Reloaded save with new mod test folder. Exception triggers on load Exception 2 - ControlAnySim Cause: Probably the buried zips again lol. But CAS identified as the cause.
> 
> Both BEs added, lastexception.txt, mc_lastexception.html are both from the second one that identifies CAS as the cause
> 
> [baby.zip](https://github.com/TitanNano/TS4ControlAnySim/files/12079774/baby.zip)

this is entirely unrelated to the mod.

**Comment by TitanNano** [comment]
Thanks for all the exceptions. The game appears to be looking for something on these sims that is not yet loaded for them (a common thing for NPC sims, the game often does not load all the data and features of NPC sims as the player can't see them anyway). Unfortunately, it's hard to tell what exactly the game is looking for, just from the exception. For further investigation, we will have to look into the game's source code. I currently don't have time for that, but I will get back to this issue when I find the time.

The exception occurs every time you try to take control of a certain type of sim (or a sim in a certain circumstance). If you have any details on what this might be, that could be useful, but don't waste your time on it, looking into the code will probably reveal the reason.

**Comment by Pirakeet** [comment]
Yeah that's perfectly understandable (re: not having time). As it is, the mod is still functioning, and control is still taken regardless (not that this would impact whether on you have time, of course)

I've stopped manually looking into it, but if I come across any I'll save it for future reference (I've started doing this for myself anyway). So feel free to let me know if/when you come back to it if you want more examples.

In the meantime I'll let you know if I figure out any details around the specific circumstances or sims. But otherwise I appreciate your time (and the creation/maintenance of this mod) :)
